src/conn.c

```diff
@@ -166,14 +183,23 @@ scheme_from_proto(int proto)
 char *
 conn_url(conn_t *conn)
 {
+	char * prefix = "";
+	char * postfix = "";
+
 	strcpy(string, scheme_from_proto(conn->proto));
 
 	if (*conn->user != 0 && strcmp(conn->user, "anonymous") != 0)
 		sprintf(string + strlen(string), "%s:%s@",
 			conn->user, conn->pass);
 
-	sprintf(string + strlen(string), "%s:%i%s%s",
```

Comment:
The URL formatting should be split to somewhere else so it can be shared between `conn.c` and `http.c`